extensions/omni_sql/omni_sql.c

```diff
@@ -583,19 +583,140 @@ Datum is_returning_statement(PG_FUNCTION_ARGS) {
 
 PG_FUNCTION_INFO_V1(execute_parameterized);
 
+/**
+  * Holds information between multiple calls to the SRF
+  * execute_parameterized function. Most of the fields are used to deal
+  * with parameters of a prepared statement.
+  */
+typedef struct ExecCtx {
+  // the actual statement to be executed.
+  char *stmt;
+  // returning is important to see if a given statement returns tuples
+  // to the caller. This controls the flow of the main function
+  bool returning;
+  // needed to finish the SRF properly (by calling SRF_RETURN_DONE)
+  bool done;
+  // used to fetch rows when the statement actually support such (see `returning`)
+  Portal portal;
+  // the number of parameters for a prepared statement
+  int nargs;
+  // the (optional) types of the parameters of the prepared statement
+  Oid *types;
+  // is any parameter null?
+  char *nulls;
+  // the arguments to replace the parameters of the prepared statement
+  Datum *values;
+} ExecCtx;
+
+/*
```

Comment:
Minor: for Doxygen (which we use in some parts of the project, but not actively), `/**` is the trigger (as opposed to `/*`)